ecommerce/products/urls.py

```diff
@@ -4,4 +4,6 @@
 
 urlpatterns = [
     path('', views.HomePageView.as_view(), name='home'),
+    path('<product_type>', views.ProductByTypeView.as_view()),
+    path('<str:id>/', views.ProductDetailView.as_view(), name='detail'),
```

Comment:
Now that you've made this `str` again there will be another clash with the `ProductByTypeView` class. I'd change it back to `int` :)

Reply:
Got it